File: Python/104.Maximum Depth of Binary Tree.py
#返回二叉树的深度
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def levelOrder(self, root):
        self.maxdepth = 0
        self.inorder(root,1)
        return self.maxdepth

    def inorder(self,p,level):
        if p!=None:
            if self.maxdepth<level:
                self.maxdepth = level
            self.inorder(p.left,level+1) if p.left!=None else None
            self.inorder(p.right,level+1) if p.right!=None else None



t1 = TreeNode(1)
t2 = TreeNode(2)
t3 = TreeNode(3)
t4 = TreeNode(4)
t5 = TreeNode(5)
t6 = TreeNode(6)
t7 = TreeNode(7)
t8 = TreeNode(8)
t1.left = t2
t1.right = t3
t2.left = t4
t2.right = t6
t3.right = t5
t3.left = t7
t7.right = t8


s = Solution()
print(s.levelOrder(t7))

File: C++/Tree/104.Maximum Depth of Binary Tree.cpp
/*
求二叉树最大深度。
*/
/*
思路：递归函数参数加一个深度即可。
 */
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution {
public:
	int maxdepth;
public:
	int maxDepth(TreeNode* root) {
		maxdepth = 0;
        if(root==NULL)	return maxdepth;
        helpLevelOrder(root,1);
        return maxdepth;
    }
    void helpLevelOrder(TreeNode* cur,int level)
    {
    	if(cur!=NULL)
    	{
    		if(level > maxdepth) maxdepth = level;
    		helpLevelOrder(cur->left,level+1);
    		helpLevelOrder(cur->right,level+1);
    	}
    }
};
int main()
{
	Solution s;
	TreeNode t1(1),t2(2),t3(3);
	t1.right = &t2;
	t2.left = &t3;
	cout << s.maxDepth(&t1) <<endl;
}

